variants/github_actions_ci/workflows/deploy_to_ec2.yml

```diff
@@ -0,0 +1,76 @@
+name: deploy_to_ec2
+on:
+  workflow_call:
+    inputs:
+      environment:
+        type: string
+        required: true
+      assume_role_arn:
+        type: string
+        required: true
+      aws_region:
+        type: string
+        required: true
+      timeout_minutes:
+        type: number
+        required: false
+        default: 15
+      slack_icon:
+        type: string
+        required: false
+        default: https://github.com/ackama.png?size=48
+      slack_channel:
+        type: string
+        required: false
+        default: TODO_CHOOSE_A_REAL_SLACK_CHANNEL
+      slack_username:
+        type: string
+        required: false
+        default: "Github Actions CI"
+    secrets:
+      ssh_private_key:
+        required: true
+      slack_webhook:
+        required: false
+jobs:
+  deploy_to_ec2_via_oidc:
+    name: Deploy to AWS EC2 with Capistrano, authenticated by Github OIDC
+    runs-on: ubuntu-latest
+    timeout-minutes: ${{ inputs.timeout_minutes }}
+    permissions:
+      # These permissions are needed to interact with GitHub's OIDC Token
+      # endpoint.
+      id-token: write
+      contents: read
+    steps:
+      - uses: actions/checkout@v3
+      - uses: ruby/setup-ruby@v1
+        with:
+          bundler-cache: true
+      - name: Configure AWS credentials using OpenID Connect to AssumeRole
+        uses: aws-actions/configure-aws-credentials@v1
+        with:
+          role-to-assume: ${{ inputs.assume_role_arn }}
+          aws-region: ${{ inputs.aws_region }}
+      - name: Install SSH key for production
+        uses: shimataro/ssh-key-action@v2
+        with:
+          key: ${{ secrets.ssh_private_key }}
+          known_hosts: unnecessary
+      - name: Deploy application
+        run: bundle exec cap ${{ inputs.environment }} deploy
+      - name: Report to Slack
+        # 2022-09-16: Github Actions cannot currently reference `secrets` in an
+        # if expression. Workaround is to copy to secret to env var. In this
+        # case we already need the secret in an env var. Details:
+        # https://github.com/actions/runner/issues/520
+        if: "${{ env.SLACK_WEBHOOK != '' }}"
```

Comment:
FYI @G-Rath

Reply:
interesting - I think we should switch on `slack_channel` then because we want that to always be provided as well and it sounds like anything used in `if` could be "leaked"

Reply:
yeah I think using `slack_channel` is what we should do here because that's optional for `rtCamp/action-slack-notify` whereas [`SLACK_WEBHOOK` is required](https://github.com/rtCamp/action-slack-notify/blob/master/main.go#L58-L62).

So providing a channel without a webhook will result in an nicer error message anyway

Reply:
inputs.slack_channel has a default value so I'd need to check it's not equal to the default which feels a bit messy

Reply:
the only reason it has a default is because we wanted it to error if a channel wasn't provided since otherwise it'll use the default channel that is set by the webhook, so the default means it instead errors about the channel not existing.

This change would mean we can get rid of that default value because we'll only be doing Slack if a channel has been provided, so we don't have to worry about the webhooks channel being used by mistake.

Reply:
Ah ok I get what you mean now. Yeah that makes sense. Will change.